zng/resolver/context.go

```diff
@@ -812,3 +813,19 @@ func (c *Context) TranslateTypeMap(ext *zng.TypeMap) (*zng.TypeMap, error) {
 	}
 	return c.LookupTypeMap(keyType, valType), nil
 }
+
+func (c *Context) NewError(format string, args ...interface{}) (zng.Value, error) {
+	msg := fmt.Sprintf(format, args...)
+	if c.typeErr == nil {
+		typ, err := c.LookupTypeAlias("error", zng.TypeString)
+		if err != nil {
+			return zng.Value{}, err
```

Comment:
Allowing this to fail causes surprising behavior if `error` is already aliased to a type other than `string`.

```sh
$ go run ./cmd/zq -t 'put i=1/i' - <<EOF
#error=bstring
#0:record[i:int64]
0:[0;]
EOF
alias exists with different type
#0:record[i:int64]
0:[0;]
```

Maybe create the `error` alias in NewContext?

An underlying matter here (and one I wish I'd seen coming) is that `error` is no longer part of the ZNG spec but is still quasi reserved by the implementation. This doesn't feel like a great place to be. Should we reserve it in the spec?